### PR TITLE
[4.0] Prepare Database for checked_out in workflow tables

### PR DESCRIPTION
**Pull request by chmst**
This pr adds the fields checked_out and checked_out_time to the workflow tables.
If this is done, the views of the workflow component can use the checked_out functions as other components. 

### Testing Instructions
Install a new application and inspect tables: workflows, workflow_stages and workflow_transitions.
Everything should work as before. 

NOTE: I could test only the mysql database Would be great if I get help with testing the postgres part.

### CODE REVIEW

**Comment by amitranjan2** [comment]
https://github.com/joomla/joomla-cms/blob/1bda2b38ff1a4c393fda50e0a8719b0817143adb/administrator/components/com_admin/sql/updates/mysql/4.0.0-2018-06-06.sql#L5 
DEFAULT should be numeric 1
 Please look at Line 6 also

**Comment by chmst** [comment]
@amitranjan2 you are right, there are also DEFAULT '0'. But this is not in scope of this PR, it is better you make an extra  PR for this.

**Comment by amitranjan2** [comment]
> @amitranjan2 you are right, there are also DEFAULT '0'. But this is not in scope of this PR, it is better you make an extra PR for this.

Okay

**Comment by HLeithner** [comment]
@alikon could you test this with pgsql?

**Comment by drmenzelit** [comment]
I have tested this item :white_check_mark: successfully on 766ac40f9771a2cd091dc59dbd2c24ff352fc149<br /><br />After applying the patch and installing Joomla! 4.0.0-alpha8-dev again I have the new columns checked_out and checked_out_time on the workflow tables.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24059">issues.joomla.org/tracker/joomla-cms/24059</a>.</sub>

**Comment by alikon** [comment]
on pgsql side seem's fine to me

**Comment by ghost** [comment]
@alikon can you please mark your test as successfully?

**Comment by VladikB** [comment]
I have tested this item :white_check_mark: successfully on 5f8e751409718909730f942bbef712e3d22469a9<br /><br />Tested it on mysql<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24059">issues.joomla.org/tracker/joomla-cms/24059</a>.</sub>

**Comment by ghost** [comment]
@drmenzelit can you please retest?

**Comment by drmenzelit** [comment]
I have tested this item :white_check_mark: successfully on 5f8e751409718909730f942bbef712e3d22469a9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24059">issues.joomla.org/tracker/joomla-cms/24059</a>.</sub>

**Comment by ghost** [comment]
Status "Ready To Commit".

File: installation/sql/postgresql/joomla.sql [hunk 1]
```diff
@@ -2325,24 +2328,25 @@ CREATE TABLE IF NOT EXISTS "#__workflow_stages" (
...
 -- Dumping data for table "#__workflow_stages"
 --
 
 INSERT INTO "#__workflow_stages" ("id", "asset_id", "ordering", "workflow_id", "published", "title", "description", "condition", "default") VALUES
-(1, 0, 1, 1, 1, 'JUNPUBLISHED', '', 0, 1),
-(2, 0, 2, 1, 1, 'JPUBLISHED', '', 1, 0),
-(3, 0, 3, 1, 1, 'JTRASHED', '', -2, 0),
-(4, 0, 4, 1, 1, 'JARCHIVED', '', 2, 0);
-
-SELECT setval('#__workflow_stages_id_seq', 5, false);
```
Comment (wilsonge) [inline installation/sql/postgresql/joomla.sql:2363]: Don't think you meant to remove this line

Comment (chmst) [reply installation/sql/postgresql/joomla.sql:2363]: fixed, thank you.

**Comment by wilsonge** [comment]
Thankyou!

**Pull request merged**